dist/samples/places-autocomplete-addressform/index.js

```diff
@@ -36,42 +48,56 @@ function initAutocomplete() {
 function fillInAddress() {
   // Get the place details from the autocomplete object.
   const place = autocomplete.getPlace();
-
-  for (const component in componentForm) {
-    document.getElementById(component).value = "";
-    document.getElementById(component).disabled = false;
-  }
+  let address1 = "";
+  let postcode = "";
 
   // Get each component of the address from the place details,
   // and then fill-in the corresponding field on the form.
   for (const component of place.address_components) {
     const addressType = component.types[0];
 
-    if (componentForm[addressType]) {
-      const val = component[componentForm[addressType]];
-      document.getElementById(addressType).value = val;
+    switch (addressType) {
+      case "street_number": {
+        address1 = component[componentLength[addressType]] + " " + address1;
+        break;
+      }
+
+      case "route": {
+        address1 += component[componentLength[addressType]];
+        break;
+      }
+
+      case "postal_code": {
+        postcode = component[componentLength[addressType]] + postcode;
+        break;
+      }
+
+      case "postal_code_suffix": {
+        postcode += "-" + component[componentLength[addressType]];
+        break;
+      }
+
+      default: {
+        if (componentLength[addressType]) {
+          const val = component[componentLength[addressType]];
+          document.getElementById(addressType).value = val;
+        }
+        break;
+      }
     }
   }
-}
+  address1Field.value = address1;
+  postalField.value = postcode;
 
-// [END maps_places_autocomplete_addressform_fillform]
-// [START maps_places_autocomplete_addressform_geolocation]
-// Bias the autocomplete object to the user's geographical location,
-// as supplied by the browser's 'navigator.geolocation' object.
-function geolocate() {
-  if (navigator.geolocation) {
-    navigator.geolocation.getCurrentPosition((position) => {
-      const geolocation = {
-        lat: position.coords.latitude,
-        lng: position.coords.longitude,
-      };
-      const circle = new google.maps.Circle({
-        center: geolocation,
-        radius: position.coords.accuracy,
-      });
-      autocomplete.setBounds(circle.getBounds());
-    });
+  // Enable the rest of the address form fields
+  for (const component of componentFields) {
+    document.getElementById(component).disabled = false;
   }
+  // After filling the form with address components from the Autocomplete
+  // prediction, set cursor focus on the second address line to encourage
+  // entry of subpremise information such as apartment, unit, or floor number.
+  address2Field.placeholder = "Apartment, unit, or floor #";
```

Comment:
why delay setting this?

Reply:
There are no placeholders in any of the fields when the form is first loaded (reduce mental load), but at this point when the cursor focus is set to this field (once the other fields are pre-filled) it is helpful to add this placeholder. It reinforces the text already shown in the field's label.

Reply:
> reduce mental load

interesting - any sources for that I can read up on?

But also, if the placeholder simply repeats the label, what value has it brought?